File: 01_Array_And_Hashing/contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        # create a hashset to store the looped numbers
        hashset = set()

        # loop through list to check duplicates
        for num in nums:
            # check if num already exists in hashset
            if num in hashset:
                return True
            # add num to hashset
            hashset.add(num)
        return False


# implementaion
print("Leetcode 217: Contains Duplicate\n")
answer = Solution()

# Example 1
example_1 = [1, 2, 3, 1]
print(f"Example 1: nums = {example_1}")
print("Output:", answer.containsDuplicate(example_1))

# Example 2
example_2 = [1, 2, 3, 4]
print(f"Example 2: nums = {example_2}")
print("Output:", answer.containsDuplicate(example_2))

# Example 3
example_3 = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2]
print(f"Example 3: nums = {example_3}")
print("Output:", answer.containsDuplicate(example_3))

File: python/01_Array_And_Hashing/contains_duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    // loop through nums list
    for (let right = 0; right < nums.length; right++) {
        // loop through each right pointer
        for (let left = 0; left < right; left++) { 
            // check if both pointers are duplicate
            const isDuplicate = nums[left] === nums[right];
            // return true if match is found
            if (isDuplicate) return true;
        }
    }
    // return false if no match 
    return false;
};


// implementation
console.log("Leetcode 217: Contains Duplicate\n");

// Example 1
const example1 = [1, 2, 3, 1];
console.log(`Example 1: num = [${example1}]`);
console.log("Output:", containsDuplicate(example1));

// Example 2
const example2 = [1, 2, 3, 4];
console.log(`Example 2: num = [${example2}]`);
console.log("Output:", containsDuplicate(example2));

// Example 3
const example3 = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2];
console.log(`Example 3: num = [${example3}]`);
console.log("Output:", containsDuplicate(example3));